Library/Formula/hbase.rb

```diff
@@ -4,32 +4,151 @@ class Hbase < Formula
   url "https://www.apache.org/dyn/closer.cgi?path=hbase/1.1.2/hbase-1.1.2-bin.tar.gz"
   sha256 "8ca5bf0203cef86b4a0acbba89afcd5977488ebc73eec097e93c592b16f8bede"
 
-  bottle :unneeded
-
   depends_on :java => "1.6+"
-  depends_on "hadoop"
+  depends_on "hadoop" => :optional
+  depends_on "lzo" => :recommended
+  depends_on "ant" => :build if build.with? "lzo"
+
+  resource "hadoop-lzo" do
+    url "https://github.com/cloudera/hadoop-lzo/archive/0.4.14.tar.gz"
+    sha256 "aa8ddbb8b3f9e1c4b8cc3523486acdb7841cd97c002a9f2959c5b320c7bb0e6c"
+  end
 
   def install
+    ENV.java_cache if build.with? "lzo"
     rm_f Dir["bin/*.cmd", "conf/*.cmd"]
     libexec.install %w[bin conf docs lib hbase-webapps]
     bin.write_exec_script Dir["#{libexec}/bin/*"]
 
-    inreplace "#{libexec}/conf/hbase-env.sh",
-      "# export JAVA_HOME=/usr/java/jdk1.6.0/",
-      "export JAVA_HOME=\"$(/usr/libexec/java_home)\""
+    if build.with? "lzo"
+      resource("hadoop-lzo").stage do
+        # Fixed upstream: https://github.com/cloudera/hadoop-lzo/blob/master/build.xml#L235
+        inreplace "build.xml",
+                  %r{(<class name="com.hadoop.compression.lzo.LzoDecompressor" />)},
+                  "\\1\n<classpath refid=\"classpath\"/>"
+        ENV["CLASSPATH"] = Dir["#{libexec}/lib/hadoop-common-*.jar"].first
+        ENV["CFLAGS"] = "-m64"
```

Comment:
Compiling without the `-m64` flags resulted in the `.a` file differing only in some timestamps, the `.la` file differing only in the record of the build directory and 

the dylib differs at this address with or without the `-m64` flag:

```
0000528 453c f8a0 1a49 32ca a996 8a5e 643c e29d
0000528 2c91 d6af 7d2d 3cbe 9e2f 9622 c684 fa33
0000528 2b81 dab4 3252 37c0 8b63 55f6 f870 ee71
```

diffing the jar's wouldn't be much use as they're compressed, but those differ too.